EntityFramework.Reverse.POCO.Generator/EF.Reverse.POCO.Core.ttinclude

```diff
@@ -3260,25 +3269,67 @@ SELECT  SERVERPROPERTY('Edition') AS Edition,
             }
 
-            public void AddReverseNavigation(Relationship relationship, string fkName, Table fkTable, string propName, string constraint, string collectionType, CommentsStyle includeComments)
+            public void AddReverseNavigation(Relationship relationship, string fkName, Table fkTable, string propName, string constraint, string collectionType, CommentsStyle includeComments, List<ForeignKey> fks, Table mappingTable=null)
             {
+                string fkNames = "";
                 switch (relationship)
                 {
                     case Relationship.OneToOne:
-                        ReverseNavigationProperty.Add(string.Format("public virtual {0} {1} {{ get; set; }}{2}", fkTable.NameHumanCaseWithSuffix, propName, includeComments != CommentsStyle.None ? " // " + constraint : string.Empty));
+                    case Relationship.OneToMany:
+                    case Relationship.ManyToOne:
+                        fkNames = string.Join(", ", fks.Select(x => "[" + x.FkColumn + "]").Distinct().ToArray());
+                        if (fks.Count>1)
+                            fkNames = "\"" + fkNames + "\"";
+                        break;
+                    case Relationship.ManyToMany:
+                        break;
+
+                    default:
+                        break;
+                }
+                switch (relationship)
+                {
+                    case Relationship.OneToOne:
+                        ReverseNavigationProperty.Add(
+                            new PropertyAndComments() 
+                            {
+                                Definition = string.Format("public virtual {0} {1} {{ get; set; }}{2}", fkTable.NameHumanCaseWithSuffix, propName, includeComments != CommentsStyle.None ? " // " + constraint : string.Empty),
+                                Comments = string.Format("Parent (One-to-One) {0} pointed by [{1}].({2}) ({3})", this.NameHumanCaseWithSuffix, fkTable.Name, fkNames, fks.First().ConstraintName)
+                            }
+                        );
                         break;
 
                     case Relationship.OneToMany:
-                        ReverseNavigationProperty.Add(string.Format("public virtual {0} {1} {{ get; set; }}{2}", fkTable.NameHumanCaseWithSuffix, propName, includeComments != CommentsStyle.None ? " // " + constraint : string.Empty));
+                        ReverseNavigationProperty.Add(
+                            new PropertyAndComments() 
+                            {
+                                Definition = string.Format("public virtual {0} {1} {{ get; set; }}{2}", fkTable.NameHumanCaseWithSuffix, propName, includeComments != CommentsStyle.None ? " // " + constraint : string.Empty),
+                                Comments = string.Format("Parent {0} pointed by [{1}].({2}) ({3})", this.NameHumanCaseWithSuffix, fkTable.Name, fkNames, fks.First().ConstraintName)
```

Comment:
I think this one is correct indeed:
"Parent Content pointed by [Content].([ParentContentUID])"
"Parent Category pointed by [Product].([CategoryID])"
Is this gramatically incorrect?